x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/LookupJoinTypesIT.java

```diff
@@ -0,0 +1,463 @@
+        );
+    }
+
+    private void initData(String group) {
+        Collection<TestConfig> configs = testConfigurations.get(group).configs.values();
+        int docId = 0;
+        for (TestConfig config : configs) {
+            String doc = String.format(Locale.ROOT, """
+                {
+                  %s,
+                  "other": "value"
+                }
+                """, lookupPropertyFor(config));
+            index(config.indexName(), "" + (++docId), doc);
+            refresh(config.indexName());
+        }
+        List<String> mainProperties = configs.stream().map(this::mainPropertyFor).distinct().collect(Collectors.toList());
+        index("index", "1", String.format(Locale.ROOT, """
+            {
+              %s
+            }
+            """, String.join(",\n  ", mainProperties)));
+        refresh("index");
+    }
+
+    private String lookupPropertyFor(TestConfig config) {
+        return String.format(Locale.ROOT, "\"%s\": %s", config.fieldName(), sampleDataTextFor(config.lookupType()));
+    }
+
+    private String mainPropertyFor(TestConfig config) {
+        return String.format(Locale.ROOT, "\"%s\": %s", config.fieldName(), sampleDataTextFor(config.mainType()));
+    }
+
+    private static String sampleDataTextFor(DataType type) {
+        return switch (type) {
+            case KEYWORD, TEXT, DATETIME, DATE_NANOS, IP -> "\"" + sampleDataFor(type) + "\"";
+            default -> String.valueOf(sampleDataFor(type));
+        };
+    }
+
+    private static Object sampleDataFor(DataType type) {
+        return switch (type) {
+            case BOOLEAN -> true;
+            case DATETIME, DATE_NANOS -> "2025-04-02T12:00:00.000Z";
+            case IP -> "127.0.0.1";
+            case KEYWORD, TEXT -> "key";
+            case BYTE, SHORT, INTEGER -> 1;
+            case LONG -> 1L;
+            case HALF_FLOAT, FLOAT, DOUBLE -> 1.0;
+            default -> throw new IllegalArgumentException("Unsupported type: " + type);
+        };
+    }
+
+    private static class TestConfigs {
+        final String group;
+        final Map<String, TestConfig> configs;
+
+        TestConfigs(String group) {
+            this.group = group;
+            this.configs = new LinkedHashMap<>();
+        }
+
+        private boolean exists(String indexName) {
+            return configs.containsKey(indexName);
+        }
+
+        private void add(TestConfig config) {
+            if (configs.containsKey(config.indexName())) {
+                throw new IllegalArgumentException("Duplicate index name: " + config.indexName());
+            }
+            configs.put(config.indexName(), config);
+        }
+
+        private void addPasses(DataType mainType, DataType lookupType) {
+            add(new TestConfigPasses(mainType, lookupType, true));
+        }
+
+        private void addEmptyResult(DataType mainType, DataType lookupType) {
+            add(new TestConfigPasses(mainType, lookupType, false));
+        }
```

Comment:
This appears unused?

Similarly, the boolean arg in the `TestConfigPasses` could always be `true`, then, no?